### PR TITLE
Add metalmatze as v0.17 release shepherd

### PR DESCRIPTION
**Pull request by metalmatze**
I'm happy to shepherd the v0.17 release in November. 

Who is doing the Sep release? :grin:

### CODE REVIEW

**Review by kakkoyun — APPROVED** [review]
✅ LGTM

File: docs/release-process.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ Release shepherd responsibilities:
 
 | Release   | Time of first RC         | Shepherd (GitHub handle) |
 |-----------|--------------------------|--------------------------|
+| v0.17.0   | (planned) 2020.11.dd     | `@metalmatze`            |
 | v0.16.0   | (planned) 2020.09.23     | Volunteer                |
 | v0.15.0   | (planned) 2020.08.12     | `@kakkoyun`              |
```
Comment (bwplotka) [inline docs/release-process.md:36]: Can you fix this? it's done already

File: docs/release-process.md [hunk 2]
```diff
@@ -31,6 +31,7 @@ Release shepherd responsibilities:
 
 | Release   | Time of first RC         | Shepherd (GitHub handle) |
 |-----------|--------------------------|--------------------------|
+| v0.17.0   | (planned) 2020.11.dd     | `@metalmatze`            |
 | v0.16.0   | (planned) 2020.09.23     | Volunteer                |
```
Comment (bwplotka) [inline docs/release-process.md:35]: Can you add me here? (:

**Review by bwplotka — APPROVED** [review]
Thanks! (:

**Pull request merged**